Fix various spelling errors.

jsoref opened the pull request:
Fix various spelling errors.

Fixes misspellings of:
* a..z (the English alphabet sequence a..z with one pair of letters transposed)
* actual
* algorithm
* amount
* although
* argument
* array
* as
* assignment
* auxiliary
* availability
* avoiding
* because
* beginning
* built
* canonicalization
* canonical
* catch
* causes
* collection
* committed
* compare
* comparison
* compilation
* completed
* concurrent
* conditions
* constructor
* contain
* context
* contiguous
* conversion
* convert
* corresponding
* counterparts
* coverage
* covered
* current
* declaration
* default
* deferred
* delimiters
* dependencies
* descriptions
* descriptor
* diagonal
* dictionary
* differently
* different
* disposed
* dynamically
* dynamic
* earlier
* emitting
* empty
* emulated
* encodable
* enough
* entrypoint
* enumerable
* environment
* evaluate
* everything
* exactly
* exception
* exercised
* existing
* expandos
* explicitly
* extension
* floatarray
* for-in
* foreign
* function
* generate
* generator
* granularity
* identical
* implemented
* implicit
* inclusive
* index
* indicate
* indicating
* infinity
* initializer
* initial
* injection
* inlinable
* inline
* insertion
* instance
* instantiate
* interior
* interpreter
* interpret
* invalidation
* invoke
* iterable
* iterator
* javascript
* label
* lambda
* lane
* language
* left
* length
* manager
* maximum
* member
* memory
* message
* metadata
* necessarily
* occurrence
* offset
* offsets
* of lit (two word sequence)
* ofobj
* ofreturn
* optimization
* original
* outer
* output
* overridable
* overridden
* owning
* parameter
* participate
* perlargest
* physical
* pollute
* polyfill
* polymorphic
* popping
* possibility
* preceded
* precise
* precision
* preferenced
* primitive
* processed
* profiled
* profile
* program
* progress
* propagate
* prop'd
* properties
* property
* prototype
* recycler
* re-enable
* referenceable
* reference
* registers
* removed
* required
* result
* semantics
* semicolon
* separately
* separator
* sequences
* should
* sibling
* signature
* similar
* simultaneously
* source
* sources
* spec'd
* specialization
* statement
* store
* straight
* subtags
* succeed
* successful
* sufficient
* summation
* supplied
* supported
* support
* surrogate
* telemetry
* ternary
* testing
* then
* the
* though
* threshold
* timing
* transferred
* traversal
* truncation
* typespec'd
* underflow
* uninitialized
* unknown
* unless
* unregister
* unroll
* unscopable
* using
* value
* variable
* verifies
* verify
* visitor
* without
* zero

brand:
* ForceES5Array

grammar:
* a/an
* any one of the ... is
* as an input/output
* because
* be replaced
* different thresholds
* have
* is
* is a
* it is
* lossless
* not to report
* operations
* required by
* the
* true
* a tortured sentence involving interleaving

Tagging an ambiguous token with a guess
* pronunciation

Whitespace

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/microsoft/chakracore/227)
<!-- Reviewable:end -->

msftclas commented:
Hi __@jsoref__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.microsoft.com.
    </span>

TTYL, MSBOT;

rajatd commented on lib/Backend/FlowGraph.h:
```diff
@@ -665,8 +665,8 @@ class Loop
         InductionVariableOpndPerUnrollMap *inductionVariableOpndPerUnrollMap;
-        // This assumes that all memop operation use the same index and has the same length
-        // Temporary map to reuse existing startIndexOpnd while emiting
+        // This assumes that all memop operation use the same index and have the same length
```
`operation` to `operations` :)

rajatd commented on lib/Parser/parse.h:
```diff
@@ -160,11 +160,11 @@ class Parser
     // This is important in Javascript because Javascript engines are required not report invalid UTF-16 sequences and to consider
-    // the UTF-16 characters pre-canonacalized. Converting this UTF-16 with invalid sequences to valid UTF-8 and back would cause
-    // all invalid UTF-16 seqences to be replace by one or more Unicode replacement characters (0xFFFD), losing the original
+    // the UTF-16 characters pre-canonical. Converting this UTF-16 with invalid sequences to valid UTF-8 and back would cause
```
I wonder if this should be `pre-canonicalization`

jsoref replied:
Well, I'm pretty sure it should be `not to report` instead of `not report` (on the previous line).

rajatd commented:
@jsoref , could you please fix the build and push another commit? Most likely has to do with not updating all the instances of a wrong spelling.

dilijev commented:
@jsoref Yep I was just looking at the logs. Definitely looks like multiple instances of variable renaming not propagating all the way through.

jsoref commented:
Sorry about those. I thought that my tools were properly replacing globally, apparently they were doing first instance only (oops). I've fixed that tool for the future.

I just wrote a tool to identify removed words (based on a diff), and I think that has identified the last stragglers. So far, it looks like there was one actual test whose output I changed (the alphabet one was using the string a couple of times), and I've now updated it.

jsoref commented:
I have a bone to pick w/ @msftclas: I've already played this game with @dnfclas here:
https://github.com/OpenLiveWriter/OpenLiveWriter/pull/134#event-491385573

How many of these bots do I have to appease?

jsoref commented:
@rajatd `operations` pushed. Thanks for reviewing. I know such things aren't the most enjoyable reading.

jsoref commented:
(admittedly, they can be rather amusing...)

rajatd commented on lib/Backend/GlobOpt.cpp:
```diff
@@ -9622,7 +9622,7 @@ GlobOpt::TypeSpecializeInlineBuiltInBinary(IR::Instr **pInstr, Value *src1Val, V
                 // Type specialize to int
                 bool retVal = this->TypeSpecializeIntBinary(pInstr, src1Val, src2Val, pDstVal, newMin, newMax, false /* skipDst */);
-                AssertMsg(retVal, "For min and max, the args have to be type-specialized to int if any both of the srces are int, but something failed during the process.");
+                AssertMsg(retVal, "For min and max, the args have to be type-specialized to int if any one of the sources are int, but something failed during the process.");
```
I think it should be `one of the sources is` instead of `are`

dilijev replied:
Agreed.

rajatd commented on lib/Backend/GlobOpt.cpp:
```diff
@@ -9632,7 +9632,7 @@ GlobOpt::TypeSpecializeInlineBuiltInBinary(IR::Instr **pInstr, Value *src1Val, V
                 src2Val = src2OriginalVal;
                 bool retVal = this->TypeSpecializeFloatBinary(instr, src1Val, src2Val, pDstVal);
-                AssertMsg(retVal, "For min and max, the args have to be type-specialized to float if any one of the src is a float, but something failed during the process.");
+                AssertMsg(retVal, "For min and max, the args have to be type-specialized to float if any one of the sources are float, but something failed during the process.");
```
same as above

dilijev commented:
@jsoref Thanks for fixing the build breaks.

rajatd commented:
@jsoref Thanks for the changes. Some of them were pretty funny :)

abchatra commented:
@jsoref  Can you please sign cla?

dilijev commented:
@jsoref sorry about the bots, but according to this one you haven't signed the CLA that applies to our project. Would you mind indulging us (to make the legal folks happy so we can merge your change)?

tcare commented:
@dilijev shouldn't CLA be picked up by all MSFT repos?

dilijev commented:
@tcare one would think, but this one is saying it's not signed? Maybe it changed recently so a new version is needed? I'm not even sure who to ask about this.

If @jsoref tries to sign the CLA at the link provided above, and it doesn't work, then there's a problem I'll try to track down. (It has worked for others.)

dilijev commented:
@tcare @bterlson since there's no new IP in this PR, I suppose we could manually bypass the CLA since it's not actually required, but since the number of lines changed is so high, I'd feel better if the bot confirmed the CLA was signed. What do you think?

tcare commented:
@dilijev @jsoref needs to make the CLA say signed, even if that means signing again.

dilijev commented:
@tcare I think the problem is that the DotNET Foundation has a different CLA than Microsoft repos.

dilijev commented:
Which is to say there's not actually a problem and once @jsoref makes @msftclas bot happy, that should be all he needs to do for our repo.

tcare commented:
Sounds good

msftclas commented:
__@jsoref__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSBOT;

dilijev commented:
Thanks for signing, @jsoref!

abchatra commented:
Thank you @jsoref

dilijev commented on test/es6/unscopablesWithScopeTest.js:
```diff
@@ -349,7 +349,7 @@ var tests = [
                 a = 3;
                 b = 4;
-                assert.areEqual(2,proto.b,"proto.b is never set b\c child b is unnscopable");
+                assert.areEqual(2,proto.b,"proto.b is never set b\c child b is unscopable");
```
Can you change 'b\\c' to 'b/c' or 'because'?

jsoref replied:
Sure, fwiw, you can see the individual changes in https://github.com/jsoref/ChakraCore/commits/spelling-full :
`because`: jsoref@de4074a (that spelling is used a bunch)
adding `is`: jsoref@8bd0db8 (and one pair of those also omitted is)

dilijev commented:
Reviewed 93 of 122 files at r1, 27 of 32 files at r2, 147 of 148 files at r3, 4 of 4 files at r4.
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[lib/Backend/GlobOpt.cpp, line 9624 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TkVqHDfs1b_kVoOAZ:-K9TkVqHDfs1b_kVoOA_:935236444) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/Backend/GlobOpt.cpp#L9624)):*
"if any one of the sources is **an** int"

---

*[lib/Backend/GlobOpt.cpp, line 9634 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TkeyepTDVj9sCK9_o:-K9TkeyepTDVj9sCK9_p:-997866385) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/Backend/GlobOpt.cpp#L9634)):*
"if any one of the sources is **a** float"

---

*[lib/common/core/FaultInjection.cpp, line 373 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TpOA8_0M3BGkgMblr:-K9TpOA8_0M3BGkgMbls:-1998367045) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/common/core/FaultInjection.cpp#L373)):*
In this case I think 'reproing' is better because we use the term 'repro' for 'reproducing an [error/bug/etc.]'

---

*[lib/common/core/FaultInjection.cpp, line 1021 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9Tp_onO8vlaNcglJzT:-K9Tp_onO8vlaNcglJzU:996262991) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/common/core/FaultInjection.cpp#L1021)):*
Same comment about 'reproing' here

---

*[lib/common/Memory/Recycler.cpp, line 6128 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9Tqy87SO6ESB8WiKuW:-K9Tqy87SO6ESB8WiKuX:270626611) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/common/Memory/Recycler.cpp#L6128)):*
@curtisman Was this supposed to be 'priority-boost--overridden' or 'priority boost overrode' or 'priority boost overrides' or something else? I'm not entirely sure this correction captures the meaning.

---

*[lib/Parser/parse.h, line 164 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TsMp4qtaqbb-gZYLD:-K9TsMp4qtaqbb-gZYLE:-768896139) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/Parser/parse.h#L164)):*
Change loss-less to lossless.

---

*[lib/Runtime/Library/InJavascript/Intl.js, line 364 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TuF59bvt17UED8KN2:-K9TuF59bvt17UED8KN3:1210285040) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/Runtime/Library/InJavascript/Intl.js#L364)):*
Be more specific that you're talking about the `"pronun"` below.

---

*[lib/Runtime/Types/DynamicObject.cpp, line 674 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TvA30ZJJIf0-bKMXG:-K9TvA30ZJJIf0-bKMXH:-1216349157) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/Runtime/Types/DynamicObject.cpp#L674)):*
I think you should change this to 're-enable' but since the meaning is clear, it's fine either way.

---

*[lib/Runtime/Types/TypeHandler.h, line 584 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TvSaj28F0EUdqVJdi:-K9TvSaj28F0EUdqVJdj:-274013065) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/Runtime/Types/TypeHandler.h#L584)):*
This should probably be `inlineSlotIndexSlot`.

---

*[test/Array/array_indexOf.js, line 79 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TvaGHdd5V5x4qkU49:-K9TvaGHdd5V5x4qkU4A:1013462882) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/test/Array/array_indexOf.js#L79)):*
These were actually fine the way they were (`arr` is an abbreviation for `array`), but this change is incorrect. If you want to change these, change to `floatArray`.

---

*[test/es5/Lex_u3.js, line 18 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9Twl8l_DillndCMJrc:-K9Twl8l_DillndCMJrd:1340431499) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/test/es5/Lex_u3.js#L18)):*
I'm guessing this comment was supposed to read `<LS><PS>`

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/microsoft/chakracore/227)*
<!-- Sent from Reviewable.io -->

dilijev commented:
@jsoref I've reviewed this entire changeset. I've left some comments on reviewable (see my previous comment).

Before I merge this in, please address those comments, and if you could, reduce this change to a smaller set of commits, and please rebase your commits on top of the latest master to help history be more legible.

I'll merge it in shortly after your next iteration.

Thanks for your work on this, and looking forward to any future PRs you've got in the works to correct similar issues.

jsoref commented:
@dilijev: so, I could easily make this one commit, I don't really know what you want in terms of commits...

dilijev commented:
@jsoref One commit is fine, and is preferred as far as we're concerned. If you had good logical reasons to split your PR into multiple commits, that would be fine too, but we just ask that you only do that if there's a clear reason.

Thanks for doing that. I'll merge this in today.

jsoref commented:
I tend to split to help maintainers review/manage risk.

If you don't care, it's certainly easy enough for me not to bother.

dilijev commented:
Reviewed 73 of 79 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[lib/Backend/IRBuilder.cpp, line 6666 \[r5\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9YEIH0q65umZKsHbNh:-K9YEIH0q65umZKsHbNi:-299500442) ([raw file](https://github.com/microsoft/chakracore/blob/c82fdf9cedf00a75573a57ed9d30dc2d01e01a9f/lib/Backend/IRBuilder.cpp#L6666)):*
There's some grammar weirdness here. You can follow up in your next PR.

---

*[lib/common/Memory/SmallFinalizableHeapBlock.cpp, line 164 \[r5\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9YEXSsrXHPcc4hwMHu:-K9YEXSsrXHPcc4hwMHv:-102366747) ([raw file](https://github.com/microsoft/chakracore/blob/c82fdf9cedf00a75573a57ed9d30dc2d01e01a9f/lib/common/Memory/SmallFinalizableHeapBlock.cpp#L164)):*
Somehow 3 spaces ended up in these comments. You can follow up later.

---

*[lib/Runtime/Library/JavascriptEnumeratorIterator.h, line 6 \[r5\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9YF1HnGMrImcu7mQ2V:-K9YF1HnGMrImcu7mQ2W:1957972955) ([raw file](https://github.com/microsoft/chakracore/blob/c82fdf9cedf00a75573a57ed9d30dc2d01e01a9f/lib/Runtime/Library/JavascriptEnumeratorIterator.h#L6)):*
I think this meant `for-in` as in `for (x in array)`

---

*[lib/Runtime/Types/PathTypeHandler.cpp, line 981 \[r5\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9YFFKvJBHpEhO1yQo5:-K9YFFKvJBHpEhO1yQo6:1932278889) ([raw file](https://github.com/microsoft/chakracore/blob/c82fdf9cedf00a75573a57ed9d30dc2d01e01a9f/lib/Runtime/Types/PathTypeHandler.cpp#L981)):*
Personal preference on this: `...have the IsPrototypeFlag set?`

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/microsoft/chakracore/227)*
<!-- Sent from Reviewable.io -->

dilijev commented:
Also previous unresolved discussions:


---

*[lib/Backend/GlobOpt.cpp, line 9624 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TkVqHDfs1b_kVoOAZ:-K9TkVqHDfs1b_kVoOA_:935236444) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/Backend/GlobOpt.cpp#L9624)):*
"if any one of the sources is **an** int"

---

*[lib/Backend/GlobOpt.cpp, line 9634 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TkeyepTDVj9sCK9_o:-K9TkeyepTDVj9sCK9_p:-997866385) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/Backend/GlobOpt.cpp#L9634)):*
"if any one of the sources is **a** float"

ianwjhalliday commented:
Yes to the for-in typo

dilijev commented:
@jsoref That's absolutely reasonable, and it can sometimes help in review to have commits split up into separate logical segments. In this particular case, my reasoning is that there are really only two kinds of changes in this PR:

1) Comment changes, which pose no risk.
2) Renaming variables in a mostly local scope, which poses only a small amount of risk, especially if the CI checks pass.

So there's only one set of "risky" changes to consider here, and any non-risky changes can be combined into it. But overall, every change in this PR is a spelling change, and so to me, it's logically a single commit.

During a code review, I personally sometimes find it helpful to have multiple smaller commits because it can act as sort of a guided tour of the review. For related logic changes in a large PR this can be very helpful, but when it comes time to merge it in, a single commit is often best, especially if you're already rebasing on top of master.

Still, it is author's discretion. If there's clear enough logical units that the author believes should be highlighted in the history, multiple commits are absolutely welcome.

jsoref commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[lib/Backend/GlobOpt.cpp, line 9624 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TkVqHDfs1b_kVoOAZ:-K9U47KnUwBDPExPi7eX:268252405) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/Backend/GlobOpt.cpp#L9624)):*
Oh, right, I was going to run through the repo and do a/an work. I've started it. It'd probably be helpful if that didn't block the rest of this. The an -> a side is fairly annoying/slow.

---

*[lib/Backend/IRBuilder.cpp, line 6666 \[r5\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9YEIH0q65umZKsHbNh:-K9YNb6_SwusWp57MS_H:829955470) ([raw file](https://github.com/microsoft/chakracore/blob/c82fdf9cedf00a75573a57ed9d30dc2d01e01a9f/lib/Backend/IRBuilder.cpp#L6666)):*
s/pass/passing/ ?? It's weird, but, it's also hard to fix, because there's so little text, and the code is fairly opaque (necessitating a comment!)

---

*[test/Array/array_indexOf.js, line 79 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TvaGHdd5V5x4qkU49:-K9UEY89pFhQ6Lmc1x0J:1134010779) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/test/Array/array_indexOf.js#L79)):*
Ok, the reason I was changing it was to have only a single spelling (both `ary` and `arr` were used), I'm aware that both are valid, but I find using a single abbreviation consistently is helpful.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/microsoft/chakracore/227)*
<!-- Sent from Reviewable.io -->

dilijev commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[lib/Backend/GlobOpt.cpp, line 9624 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TkVqHDfs1b_kVoOAZ:-K9YPDXBInA3G91b8D4u:1157050218) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/Backend/GlobOpt.cpp#L9624)):*
Yeah, I don't want to block on this. I assumed you may have been working on additional changes in your other branch and I assume there will be another nice big spelling PR coming in.

Still these were omissions rather than a/an conversions and since I noticed them I figured I'd point them out. Feel free to address them in your other branch and I'll finish up merging in this PR.

---

*[lib/Backend/IRBuilder.cpp, line 6666 \[r5\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9YEIH0q65umZKsHbNh:-K9YPghTWKK2U_MY-Dg9:-1654542451) ([raw file](https://github.com/microsoft/chakracore/blob/c82fdf9cedf00a75573a57ed9d30dc2d01e01a9f/lib/Backend/IRBuilder.cpp#L6666)):*
Hmm, yeah. Maybe just leave this one alone for now. We can consider it in your future PR. Maybe @curtisman knows more about the context here?

---

*[test/Array/array_indexOf.js, line 79 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TvaGHdd5V5x4qkU49:-K9YPwWaDRTzLZH0tG4z:-1429900686) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/test/Array/array_indexOf.js#L79)):*
Ah, I see. It's such a small difference for the abbreviation I think spelling it out is better. Thanks for the change you made.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/microsoft/chakracore/227)*
<!-- Sent from Reviewable.io -->

dilijev commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[lib/Runtime/Library/JavascriptEnumeratorIterator.h, line 6 \[r5\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9YF1HnGMrImcu7mQ2V:-K9YQ9ZIn-ONz82eIEIK:-1854444983) ([raw file](https://github.com/microsoft/chakracore/blob/c82fdf9cedf00a75573a57ed9d30dc2d01e01a9f/lib/Runtime/Library/JavascriptEnumeratorIterator.h#L6)):*
This one should definitely be changed in this PR.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/microsoft/chakracore/227)*
<!-- Sent from Reviewable.io -->

jsoref commented:
Sure. From my perspective, if you're comfortable with it completely folded, I'm fine with it.
Comfort varies a lot by project and test coverage.
Some projects have API/ABI rules which means that changing a function name or commandline argument is more problematic than others. So I try to be cautious about how I deliver changes.

For TypeScript and another Microsoft project, I'm going to try tossing the series raw and see what they want instead of spending the time categorizing.

Note that I benefit from playing around w/ all of these segmentations, since I'm working on improving Mercurial (hg)'s UX, which means I'm forced to use more tools/paths and find bugs/improve edges as I go.

---

Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[lib/Backend/GlobOpt.cpp, line 9624 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9TkVqHDfs1b_kVoOAZ:-K9YQp_v1wdRLLl9Wocq:29774154) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/Backend/GlobOpt.cpp#L9624)):*
https://github.com/jsoref/ChakraCore/commit/685932a28dd40270df0276493e4ede617c6a20c8 has the only changes I'm holding off.

---

*[lib/Backend/IRBuilder.cpp, line 6666 \[r5\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9YEIH0q65umZKsHbNh:-K9YRzEeCHwEfKNz4YIJ:-744729709) ([raw file](https://github.com/microsoft/chakracore/blob/c82fdf9cedf00a75573a57ed9d30dc2d01e01a9f/lib/Backend/IRBuilder.cpp#L6666)):*
 I can stash a `passing` change into my pending branch for once everything else goes in.

---

*[lib/common/Memory/Recycler.cpp, line 6128 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9Tqy87SO6ESB8WiKuW:-K9YR-aMMux1CCAjdfJ2:27090597) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/common/Memory/Recycler.cpp#L6128)):*
Do we want me to keep an open thing for this?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/microsoft/chakracore/227#-:-K9YRGh2GLlrWrLH2esj:1191977792)*
<!-- Sent from Reviewable.io -->

dilijev commented:
Cool. In the days before I was active on GitHub I used Mercurial a lot. Good to see there's an eye to continued improvement. (Of course that's totally off-topic =P)

---

Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[lib/common/Memory/Recycler.cpp, line 6128 \[r3\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9Tqy87SO6ESB8WiKuW:-K9YSXlupp1rxazJ5flD:1961429978) ([raw file](https://github.com/microsoft/chakracore/blob/f0c2f7c8faa371f6a9b8ade3a4a37d30d0e16763/lib/common/Memory/Recycler.cpp#L6128)):*
Considering the spelling changes are low-impact and there's no harm in making no change, I wouldn't worry about this unless it bothers you. I don't want to block on this.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/microsoft/chakracore/227#-:-K9YSltVEe3MOwBYH6BA:2099173144)*
<!-- Sent from Reviewable.io -->

dilijev commented:
One thing to note about Git is that it handles tracking of file renames best if the rename does not include any other changes to the file in the same commit. If you wanted to split those out, there's a good case for that. Of course if you didn't make any other changes within those files, no need.

jsoref commented:
Grr. Yeah, I wondered about that. Since mercurial tracks renames as an action, that's less of a problem for me. I'm pretty sure there are changes in the renamed files.
I'm not quite sure how to deal with them, since the files reference themselves. Would you rather files renamed before or after tests or both?

* `all changes` `rename files`
* `changes except file references` `rename files` `fix references to files`
* `rename files` `all changes`

dilijev commented:
Yep I confirmed that git-blame now lists you as the author of the whole file. Before you rebased, it had the right info. So somehow that worked out...

I think option 2 is the best one.

Luckily we don't need to worry about intermediate steps being broken due to the PR-and-merge process. There's a good case for it here.

jsoref commented:
Review status: 291 of 298 files reviewed at latest revision, 3 unresolved discussions.

---

*[lib/Backend/IRBuilder.cpp, line 6666 \[r5\]](https://reviewable.io:443/reviews/microsoft/chakracore/227#-K9YEIH0q65umZKsHbNh:-K9YYDUjSu-aCwM8YMnn:-18384944) ([raw file](https://github.com/microsoft/chakracore/blob/c82fdf9cedf00a75573a57ed9d30dc2d01e01a9f/lib/Backend/IRBuilder.cpp#L6666)):*
@curtisman I don't suppose it should be `skipping` instead?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/microsoft/chakracore/227)*
<!-- Sent from Reviewable.io -->

jsoref commented:
Alright, it'll take a minute or three, but no big deal. Again, I get to test tooling :)

dilijev commented:
Oh weird, actually the reason it's showing up as you is because after your rebase the rename wasn't preserved as a rename and you've simply added those files this time.

See:
```
ChakraCore (jsoref-spelling)
$ git ls-files | grep -i watson
lib/common/Memory/RecyclerWatsonTelemetry.h
lib/common/Memory/RecyclerWatsonTelemtry.h
```

I think git will use some heuristic on how much of a file changed in order to determine a rename. It doesn't need to match 100% but like 90%+ and you should be okay.

jsoref commented:
Yeah. I noticed that when I looked at one of my diff reviews, I was wondering if I lost the info... *sigh*, I'll need to see if my blackbox shows where it got lost...

dilijev commented:
Reviewed 7 of 7 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/microsoft/chakracore/227)*
<!-- Sent from Reviewable.io -->

dilijev commented:
Once you get the file rename info re-integrated I'll sign off and merge this PR. Thanks for bearing with me through this :)

dilijev commented:
> Some projects have API/ABI rules which means that changing a function name or commandline argument is more problematic than others. So I try to be cautious about how I deliver changes.

This is a good policy. ChakraCore will have some of these limitations too when it comes to any function names / APIs that are exported, and possibly in JIT code generation or bytecode names. In this case, I don't think anything like that applied.

dilijev commented:
> +1,022 −1,021

***That*** looks right.

jsoref commented:
The +1 is the `pronun` comment.

jsoref commented:
One change that did happen is some error messages changed. Sometimes people (as in the tests) look for very specific error strings.

That said, I'm not sure how much sympathy should be allocated to that... Especially since each js engine is generally free to have its own error strings, or to localize them, resulting in far ranging messages.

jsoref commented:
@dilijev : Do I need to work around the rename, or is git now happy w/ me?

dilijev commented:
Looks like git is happy. I'll merge your change.

ianwjhalliday commented:
For the curious, the reason our tests look for specific error messages is so that we have better certainty that we are getting the expected error in our test, and not some other error with the same constructor.

dilijev commented:
@jsoref Looking for specific messages may be the easiest way to check whether a particular error path has been hit which is specific to our engine and our code base anyway, so I don't see a problem there. As long as all the tests have been updated properly it looks like the right thing to do.

jsoref commented:
@ianwjhalliday , @dilijev : yeah, I'm not saying your tests were written wrong, it was definitely the right way to write them, just noting that someone could be incredibly skittish and refuse to change even error messages -- because it could break code written expecting those messages. I'm glad you guys aren't that conservative :)

dilijev commented:
@jsoref It remains to be seen what might happen with internal testing (which I've just started). If there are any breakages, I'm sure it will be as easy to fix them as changing some more test messages, but I doubt there will be problems.

dilijev commented:
@jsoref FYI on the merge commit

```
 rename lib/common/Memory/{RecyclerWatsonTelemtry.h => RecyclerWatsonTelemetry.h} (100%)
 rename test/InlineCaches/{inlnieLocalCacheWithoutProperty.baseline => inlineLocalCacheWithoutProperty.baseline} (100%)
 rename test/InlineCaches/{inlnieLocalCacheWithoutProperty.js => inlineLocalCacheWithoutProperty.js} (100%)
 rename test/Object/{propertyIsEnumarable.js => propertyIsEnumerable.js} (100%)
 rename test/es5/{SemiCollonAfterBlockEs5.baseline => SemicolonAfterBlockEs5.baseline} (100%)
 rename test/es5/{SemiCollonAfterBlockEs5.js => SemicolonAfterBlockEs5.js} (100%)
```

dilijev commented:
@jsoref could you change your initial comment on this pull request to include any additional information you'd like to end up in the git history log (like which words were corrected, and so on).

Also please don't make any other changes to the commits in this PR as I'm integrating the changes into the internal code, which references some things that changed, and that will just make things more difficult.

jsoref commented:
Something like:

```
Fix various spelling errors.

Fixes misspellings of:
* a..z (the English alphabet sequence a..z with one pair of letters transposed)
* actual
* algorithm
* amount
* although
* argument
* array
* as
* assignment
* auxiliary
* availability
* avoiding
* because
* beginning
* built
* canonicalization
* canonical
* catch
* causes
* collection
* committed
* compare
* comparison
* compilation
* completed
* concurrent
* conditions
* constructor
* contain
* context
* contiguous
* conversion
* convert
* corresponding
* counterparts
* coverage
* covered
* current
* declaration
* default
* deferred
* delimiters
* dependencies
* descriptions
* descriptor
* diagonal
* dictionary
* differently
* different
* disposed
* dynamically
* dynamic
* earlier
* emitting
* empty
* emulated
* encodable
* enough
* entrypoint
* enumerable
* environment
* evaluate
* everything
* exactly
* exception
* exercised
* existing
* expandos
* explicitly
* extension
* floatarray
* for-in
* foreign
* function
* generate
* generator
* granularity
* identical
* implemented
* implicit
* inclusive
* index
* indicate
* indicating
* infinity
* initializer
* initial
* injection
* inlinable
* inline
* insertion
* instance
* instantiate
* interior
* interpreter
* interpret
* invalidation
* invoke
* iterable
* iterator
* javascript
* label
* lambda
* lane
* language
* left
* length
* manager
* maximum
* member
* memory
* message
* metadata
* necessarily
* occurrence
* offset
* offsets
* of lit (two word sequence)
* ofobj
* ofreturn
* optimization
* original
* outer
* output
* overridable
* overridden
* owning
* parameter
* participate
* perlargest
* physical
* pollute
* polyfill
* polymorphic
* popping
* possibility
* preceded
* precise
* precision
* preferenced
* primitive
* processed
* profiled
* profile
* program
* progress
* propagate
* prop'd
* properties
* property
* prototype
* recycler
* re-enable
* referenceable
* reference
* registers
* removed
* required
* result
* semantics
* semicolon
* separately
* separator
* sequences
* should
* sibling
* signature
* similar
* simultaneously
* source
* sources
* spec'd
* specialization
* statement
* store
* straight
* subtags
* succeed
* successful
* sufficient
* summation
* supplied
* supported
* support
* surrogate
* telemetry
* ternary
* testing
* then
* the
* though
* threshold
* timing
* transferred
* traversal
* truncation
* typespec'd
* underflow
* uninitialized
* unknown
* unless
* unregister
* unroll
* unscopable
* using
* value
* variable
* verifies
* verify
* visitor
* without
* zero

brand:
* ForceES5Array

grammar:
* a/an
* any one of the ... is
* as an input/output
* because
* be replaced
* different thresholds
* have
* is
* is a
* it is
* lossless
* not to report
* operations
* required by
* the
* true
* a tortured sentence involving interleaving

Tagging an ambiguous token with a guess
* pronunciation

Whitespace
```

dilijev commented:
@jsoref sure, LGTM. You can edit the PR description or I'll copy-paste that up there.

jsoref commented:
I've edited the commit message (I'm not sure if that's what you wanted me to do).

jsoref commented:
If it isn't, please just do the magic. Github is somewhat befuddling.

dilijev commented:
@jsoref I meant to edit the description of the PR on GitHub itself. But editing the commit message as well does no harm.

abchatra commented:
@dilijev Is this ready to get merged?

dilijev commented:
@abchatra yes, but due to the recent issues with the internal build system I haven't been able to get a clean build yet. I'm trying every so often.

dilijev commented:
@jsoref Thanks for the spelling changes! If you have any other related changes, feel free to open a new PR when ready. :)
